sharding/contracts/validator_manager.v.py

```diff
@@ -322,3 +326,95 @@ def add_header(
     )
 
     return True
+
+
+# Helper function to get vote count
+@public
+@constant
+def get_vote_count(shard_id: int128) -> int128:
+    current_vote_in_uint: uint256 = convert(self.current_vote[shard_id], 'uint256')
+
+    # Extract current vote count(last byte of current_vote)
+    return convert(
+        uint256_mod(
+            current_vote_in_uint,
+            convert(
+                2**8,
+                'uint256'
+            )
+        ),
+        'int128'
+    )
+
+
+# Helper function to get vote count
+@public
+@constant
+def if_notary_has_vote(shard_id: int128, index: int128) -> bool:
+    # Right shift current_vote then AND(bitwise) it's value with value 1 to see if
+    # notary of given index in bitfield had voted.
+    current_vote_in_uint: uint256 = convert(self.current_vote[shard_id], 'uint256')
+    # Convert the result from integer to bool
+    return not not bitwise_and(
+        current_vote_in_uint,
+        shift(convert(1, 'uint256'), 255 - index),
+    )
+
+
+@private
+def update_vote(shard_id: int128, index: int128) -> bool:
+    current_vote_in_uint: uint256 = convert(self.current_vote[shard_id], 'uint256')
+    index_in_bitfield: uint256 = shift(convert(1, 'uint256'), 255 - index)
+
+    # Update bitfield
+    current_vote_in_uint = bitwise_or(current_vote_in_uint, index_in_bitfield)
+    # Update vote count
+    current_vote_in_uint = uint256_add(current_vote_in_uint, convert(1, 'uint256'))
+    self.current_vote[shard_id] = convert(current_vote_in_uint, 'bytes32')
+
+    return True
```

Comment:
It seems the return value is not necessary in this function

Reply:
Might be helpful for local call?